### PR TITLE
Added API server load balancers HA requirement for production use

### PR DESCRIPTION
**Pull request by acomabon**
Based on BZ #1567868, need to have this in the official documentation. It applies to all OCP versions. @openshift/team-documentation

### CODE REVIEW

File: architecture/infrastructure_components/kubernetes_infrastructure.adoc [hunk 1]
```diff
@@ -147,6 +147,16 @@ following availability:
 |Balances load between API master endpoints
 |===
 
+[NOTE]
+====
+The API Server load balancer HA must be ensured for production use. The unavailability of the
```
Comment (kalexand-rh) [inline architecture/infrastructure_components/kubernetes_infrastructure.adoc]: @acomabon, how does this sound? 

In production {product-title} clusters, you must maintain high unavailability
of the API Server load balancer. If the API Server load balancer is not
available, nodes cannot report their status, all their pods are marked dead,
and the pods' endpoints are removed from the service.

You must separately configure HA for the API Server load balancer. To configure
HA for it, you can run multiple HAProxy load balancers and use Keepalived to 
provide a constant virtual IP address for them.

**Review by kalexand-rh — CHANGES_REQUESTED** [review]

File: architecture/infrastructure_components/kubernetes_infrastructure.adoc (hunk 1 above)
Comment (acomabon) [reply architecture/infrastructure_components/kubernetes_infrastructure.adoc]: Thanks for your fast reply!
I'd say:

"In production {product-title} clusters, you must maintain high availability (not unavailability)
of the API Server load balancer...."

and in the last sentence:

"you can run multiple HAProxy load balancers and use Keepalived to
provide a floating (better word than constant) virtual IP address for them."

What do you think?

Comment (kalexand-rh) [reply architecture/infrastructure_components/kubernetes_infrastructure.adoc]: I'm happy to help.
Both your suggestions sound good to me. :) Thank you for spotting my unfortunate typo!

**Comment by acomabon** [comment]
Changes submitted.

**Review by kalexand-rh — APPROVED** [review]
LGTM :fireworks: 
Thanks!

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL

**Comment by kalexand-rh** [comment]
Rebased on https://github.com/openshift/openshift-docs/pull/10132. I'll get a peer review and merge it there. Thank you @acomabon

**Pull request closed without merging**